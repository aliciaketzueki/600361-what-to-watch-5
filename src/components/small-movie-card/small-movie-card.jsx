import React from "react";
import {Link} from "react-router-dom";
import Video from "../video/video";
import {validShape, validFunc, validOneOfType} from "../../utils/props";

const SmallMovieCard = (props) => {
  const {film, onMouseEnter, onMouseLeave, video} = props;
  const {name} = film;

  return (
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      <div className="small-movie-card__image">
        <Video film={film} video={video} />
      </div>
      <h3 className="small-movie-card__title">
        <Link to="/films/:id" className="small-movie-card__link">{name}</Link>
      </h3>
    </article>
  );
};

SmallMovieCard.propTypes = {
  film: validShape,
  video: validOneOfType,
  onMouseEnter: validFunc,
  onMouseLeave: validFunc
};

export default SmallMovieCard;
