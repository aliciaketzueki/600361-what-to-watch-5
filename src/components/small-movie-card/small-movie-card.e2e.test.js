import React from "react";
import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SmallMovieCard from "./small-movie-card";
import {film, noop} from "./small-movie-card.test";

configure({adapter: new Adapter()});

describe(`Hover SmallMovieCard`, () => {
  // Наведение на карточку фильма
  it(`Should SmallMovieCard be mouseentered`, () => {
    const handleMouseEnter = jest.fn();

    const wrapper = shallow(
        <SmallMovieCard
          film={film}
          onMouseEnter={handleMouseEnter}
          onMouseLeave={noop}
        />
    );

    wrapper.find(`.small-movie-card`).simulate(`mouseenter`);
    expect(handleMouseEnter).toHaveBeenCalledTimes(1);
  });

  // Уведение мыши с карточки фильма
  it(`Should SmallMovieCard be mouseleave`, () => {
    const handleMouseLeave = jest.fn();

    const wrapper = shallow(
        <SmallMovieCard
          film={film}
          onMouseEnter={noop}
          onMouseLeave={handleMouseLeave}
        />
    );

    wrapper.find(`.small-movie-card`).simulate(`mouseleave`);
    expect(handleMouseLeave).toHaveBeenCalledTimes(1);
  });
});
