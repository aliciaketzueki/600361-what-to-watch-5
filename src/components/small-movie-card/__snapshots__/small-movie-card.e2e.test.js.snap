// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SmallMovieCard Should SmallMovieCard render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      muted={true}
      playsInline={true}
      preload="auto"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      The Grand Budapest Hotel
    </a>
  </h3>
</article>
`;
